crates/symsorter/src/utils.rs

```diff
@@ -26,31 +26,30 @@ pub fn is_bundle_id(input: &str) -> bool {
 }
 
 /// Gets the unified ID from an object.
-pub fn get_unified_id(obj: &Object) -> String {
+pub fn get_unified_id(obj: &Object) -> Result<String> {
     if obj.file_format() == FileFormat::Pe || obj.code_id().is_none() {
-        obj.debug_id().breakpad().to_string().to_lowercase()
+        let debug_id = obj.debug_id();
+        if debug_id.is_nil() {
+            Err(anyhow!("failed to generate debug identifier"))
+        } else {
+            Ok(debug_id.breakpad().to_string().to_lowercase())
+        }
     } else {
-        obj.code_id().as_ref().unwrap().as_str().to_string()
+        Ok(obj.code_id().as_ref().unwrap().as_str().to_string())
```

Comment:
Could you check that this identifier is non-empty, too?

Reply:
Good catch, this also tripped me up as well when I was working on this: L30 checks this (`|| obj.code_id().is_none()`) and enters the debug ID block if `.code_id()` is empty.

I was thinking of ways to make it more obvious that this `else` block runs with a non-empty `code_id`, but anything that came to mind felt like an overcomplication. For example, a match block was one of my ideas:

```rust
match (obj.file_format(), obj.code_id()) {
    (Pe, _) 
    | (_, None) => use debug_id,
    (_, _) => use code
}
```

Thoughts, opinions?

Reply:
Heh, you know our code better than we do. An in fact, `code_id` can also never be empty since it always returns `None` instead. This is good to go then 👍